core/src/main/java/org/elasticsearch/search/aggregations/pipeline/movavg/models/MovAvgModel.java

```diff
@@ -238,12 +243,19 @@ protected boolean parseBoolParam(@Nullable Map<String, Object> settings, String
             if (value == null) {
                 return defaultValue;
             } else if (value instanceof Boolean) {
+                settings.remove(name);
                 return (Boolean)value;
             }
 
             throw new ParseException("Parameter [" + name + "] must be a boolean, type `"
                     + value.getClass().getSimpleName() + "` provided instead", 0);
         }
+
+        protected void checkUnrecognizedParams(@Nullable Map<String, Object> settings) throws InvalidArgumentException {
+            if (settings != null && settings.size() > 0) {
+                throw new IllegalArgumentException("Unrecognized parameter(s): [" + String.join(", ", settings.keySet()) + "]");
```

Comment:
I think this should be a ParseException instead of an IllegalArgumentException. This would be in line with what we do elsewhere in the codebase (such as in the mapper parsers where any left over parameters after parsing is done are throw in a MapperParsingException).